installation of simplejson breaks airlfow webserver 2.0.0rc2

Version 2.0.0rc2

To reproduce:
1. `pip install apache-airflow==2.0.0rc2`
2. `pip install simplejson`
3. run webserver
4. open in browser and observe following error

Error:
```
[2020-12-13 11:37:28 -0800] [85061] [INFO] Starting gunicorn 19.10.0
[2020-12-13 11:37:28 -0800] [85061] [INFO] Listening at: http://0.0.0.0:8080 (85061)
[2020-12-13 11:37:28 -0800] [85061] [INFO] Using worker: sync
[2020-12-13 11:37:28 -0800] [85064] [INFO] Booting worker with pid: 85064
[2020-12-13 11:37:36,444] {app.py:1892} ERROR - Exception on /home [GET]
Traceback (most recent call last):
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask/app.py", line 2446, in wsgi_app
    ctx.push()
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask/ctx.py", line 390, in push
    self.session = session_interface.open_session(self.app, self.request)
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask/sessions.py", line 340, in open_session
    s = self.get_signing_serializer(app)
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask/sessions.py", line 336, in get_signing_serializer
    signer_kwargs=signer_kwargs,
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/itsdangerous/serializer.py", line 95, in __init__
    self.is_text_serializer = is_text_serializer(serializer)
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/itsdangerous/serializer.py", line 13, in is_text_serializer
    return isinstance(serializer.dumps({}), text_type)
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask/json/tag.py", line 305, in dumps
    return dumps(self.tag(value), separators=(",", ":"))
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask/json/__init__.py", line 211, in dumps
    rv = _json.dumps(obj, **kwargs)
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/simplejson/__init__.py", line 412, in dumps
    **kw).encode(obj)
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/airflow/utils/json.py", line 36, in __init__
    super().__init__(*args, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'encoding'
[2020-12-13 11:37:36 -0800] [85064] [ERROR] Error handling request /home
Traceback (most recent call last):
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask/app.py", line 2446, in wsgi_app
    ctx.push()
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask/ctx.py", line 390, in push
    self.session = session_interface.open_session(self.app, self.request)
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask/sessions.py", line 340, in open_session
    s = self.get_signing_serializer(app)
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask/sessions.py", line 336, in get_signing_serializer
    signer_kwargs=signer_kwargs,
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/itsdangerous/serializer.py", line 95, in __init__
    self.is_text_serializer = is_text_serializer(serializer)
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/itsdangerous/serializer.py", line 13, in is_text_serializer
    return isinstance(serializer.dumps({}), text_type)
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask/json/tag.py", line 305, in dumps
    return dumps(self.tag(value), separators=(",", ":"))
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask/json/__init__.py", line 211, in dumps
    rv = _json.dumps(obj, **kwargs)
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/simplejson/__init__.py", line 412, in dumps
    **kw).encode(obj)
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/airflow/utils/json.py", line 36, in __init__
    super().__init__(*args, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'encoding'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/gunicorn/workers/sync.py", line 135, in handle
    self.handle_request(listener, req, client, addr)
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/gunicorn/workers/sync.py", line 176, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask/app.py", line 2464, in __call__
    return self.wsgi_app(environ, start_response)
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask/app.py", line 2450, in wsgi_app
    response = self.handle_exception(e)
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask/app.py", line 1879, in handle_exception
    server_error = handler(server_error)
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/airflow/www/views.py", line 372, in show_traceback
    else 'Error! Please contact server admin.',
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask/templating.py", line 136, in render_template
    ctx.app.update_template_context(context)
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask/app.py", line 838, in update_template_context
    context.update(func())
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask_login/utils.py", line 368, in _user_context_processor
    return dict(current_user=_get_user())
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask_login/utils.py", line 335, in _get_user
    current_app.login_manager._load_user()
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/flask_login/login_manager.py", line 346, in _load_user
    is_missing_user_id = 'user_id' not in session
  File "/Users/dstandish/.pyenv/versions/3.7.5/lib/python3.7/site-packages/werkzeug/local.py", line 379, in <lambda>
    __contains__ = lambda x, i: i in x._get_current_object()
TypeError: argument of type 'NoneType' is not iterable
127.0.0.1 - - [13/Dec/2020:11:37:36 -0800] "GET /home HTTP/1.1" 500 0 "-" "-"
^C[2020-12-13 11:37:38,288] {webserver_command.py:430} INFO - Received signal: 2. Closing gunicorn.
[2020-12-13 11:37:38 -0800] [85061] [INFO] Handling signal: int
[2020-12-13 11:37:38 -0800] [85064] [INFO] Worker exiting (pid: 85064)
[2020-12-13 11:37:38 -0800] [85061] [INFO] Shutting down: Master
```

--- a/airflow/utils/json.py
+++ b/airflow/utils/json.py
@@ -16,10 +16,10 @@
 # specific language governing permissions and limitations
 # under the License.
 
-import json
 from datetime import date, datetime
 
 import numpy as np
+from flask.json import JSONEncoder
 
 try:
     from kubernetes.client import models as k8s
@@ -29,7 +29,7 @@ except ImportError:
 # Dates and JSON encoding/decoding
 
 
-class AirflowJsonEncoder(json.JSONEncoder):
+class AirflowJsonEncoder(JSONEncoder):
     """Custom Airflow json encoder implementation."""
 
     def __init__(self, *args, **kwargs):

